data_visualization_in_seaborn/data_visualization_seaborn.md

```diff
@@ -0,0 +1,684 @@
+
+For more background on `regplot` vs. `lmplot`, see [the seaborn regression tutorial](https://seaborn.pydata.org/tutorial/regression.html#controlling-the-size-and-shape-of-the-plot).
+</div>
+
+```python
+sns.lmplot(data = covid_data,
+            x="val_age", y="val_height_cm")
+```
+
+![Scatterplot with age on the x-axis and height on the y-axis, and a straight trend line running through the points, with a lighter shaded band around it.](media/seaborn_trend_1.png)
+
+A couple things about this plot are not ideal:
+
+1. Overplotting. The density of the scatter plot makes it hard to see the trend line as it runs through the bulk of the data.
+2. The linear trend doesn't appear to be a very good description of the relationship between age and height. It systematically over-estimates height at the very young ages, then under-estimates through early adulthood and over-estimates again late in life.
+
+We can address the first issue by adjusting alpha for the scatterplot. Alpha ranges between 0 (totally transparent) and 1 (totally opaque).
+
+<div class = "important">
+Tip: When your data overlap too much on a plot, use alpha to make them more transparent.
+</div>
+
+We face a problem here, though: Since we're using a single function to draw both the scatter plot and the trend line, how do we adjust alpha for the scatter plot without affecting the line?
+
+The solution is to pass additional arguments to the `matplotlib` functions on which `seaborn` is built.
+
+Although `seaborn` is great at making many complex data visualizations smoothly and with very few lines of code, it sometimes lacks control of little details within those plots. It is built on top of the extremely powerful and flexible `matplotlib` library, however, so if you ever find yourself wanting more control of your `seaborn` visualizations, the answer is often to leverage the underlying `matplotlib` functions.
+
+<div class = "care">
+There is a much steeper learning curve for `matplotlib` compared to `seaborn`, however, and you may find you're happy with most `seaborn` plots without needing additional control --- don't bother with it if it feels too overwhelming. When you're ready to read more about `matplotlib`, start with their [introductory tutorials](https://matplotlib.org/stable/tutorials/index).
+</div>
+
+In this case, we can add an argument `scatter_kws` that passes additional "keyword arguments" to the `matplotlib` function drawing the scatter plot via a [python dictionary](https://www.w3schools.com/python/python_dictionaries.asp). There are [many possible keyword arguments](https://matplotlib.org/3.5.1/api/_as_gen/matplotlib.lines.Line2D.html#matplotlib.lines.Line2D) you can use, but we'll just set `alpha` here.
```

Comment:
is it worthwile to explain what the scatter_kws argument is and why we pass it things in a dictionary, or will that get way too into the weeds very quickly?

Reply:
Haha yeah, I was thinking that was perhaps too weedy. The links provided will guide folks who want to dig further into it, but I think for the actual module content it makes sense to just focus on how to implement this particular change (adjusting alpha) rather than opening the pandora's box of matplotlib arguments.